### ISSUE TITLE
Broken test

### ISSUE THREAD

**Issue by hlamer** [bjones1/enki#41]
Hi

I've discovered, that commit
```
3148469e1839dafcad17d3d229530b553944a61c is the first bad commit
commit 3148469e1839dafcad17d3d229530b553944a61c
Author: Bryan A. Jones <bjones AT ece.msstate.edu>
Date:   Wed Mar 4 21:03:22 2015 -0600

    Fix: As in previous commit, make sure all preview sync run in the main loop.
```
breaks one of the tests for me:

```
Traceback (most recent call last):
  File "/home/andrei/code/enki/tests/test_plugins/../base.py", line 97, in execWithArgs
    func(*args)
  File "./test_preview_sync.py", line 257, in test_sync12
    text after table""", False)
  File "./test_preview_sync.py", line 219, in _textToWeb
    self._dock().previewSync.textToPreviewSynced, 350)
  File "/home/andrei/code/enki/tests/test_plugins/../base.py", line 394, in assertEmits
    timeoutMs, expectedSignalParams))
  File "/usr/lib/python2.7/unittest/case.py", line 424, in assertTrue
    raise self.failureException(msg)
AssertionError: False is not true
```

**Comment by bjones1** [comment]
I was able to reproduce this on Windows the first time I executed `run_all.py`, but not after that. I'll check on Fedora 21...

**Comment by hlamer** [comment]
Linux Mint 16 64 bit

**Comment by bjones1** [comment]
Thanks. Will test.

**Comment by hlamer** [comment]
Not reproducible on Mint 17.1 on slower machine

**Comment by bjones1** [comment]
I reproduced this on Mint x64 16. I'll work on it. I have several deadlines this week, so progress will be slow.

**Comment by bjones1** [comment]
Pan, would you take a look at failures in test_previewCheck19, 20, 21? Seems to be related to spaces in path names in advanced mode.

**Comment by panhejia** [comment]
I am testing using mint 16 64bit. I cannot reproduce any of those bugs. Instead when I run ``run_all.py`` I get the following:
```
======================================================================
FAIL: test_4 (test_core.test_uisavefiles.Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/minttester/enki/tests/test_lib/../base.py", line 314, in <lambda>
    QTimer.singleShot(20, lambda: timerCallback(attempt + 1))
  File "/home/minttester/enki/tests/test_lib/../base.py", line 316, in timerCallback
    self.fail("Dialog not found")
AssertionError: Dialog not found

======================================================================
FAIL: test_5 (test_core.test_uisavefiles.Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/minttester/enki/tests/test_lib/../base.py", line 314, in <lambda>
    QTimer.singleShot(20, lambda: timerCallback(attempt + 1))
  File "/home/minttester/enki/tests/test_lib/../base.py", line 316, in timerCallback
    self.fail("Dialog not found")
AssertionError: Dialog not found

----------------------------------------------------------------------
Ran 275 tests in 122.601s

FAILED (failures=2, skipped=66)
```

I am testing enki based on commit 24e07411 .

**Comment by bjones1** [comment]
The errors I'm seeing relate to CodeChat failing under the new Sphinx 1.3.x. However, I can't reproduce the original test failures. Closing until they can be reproduced.

**Comment by hlamer** [comment]
The fail is reproduced stable for me.
How can I help you to reproduce it?
(Can not give VM image, since it is not a VM)

```
andrei@a-desktop ~/code/enki/tests/test_plugins (master)*$ pip freeze | grep CodeChat 
CodeChat==0.0.18
```

```
andrei@a-desktop ~/code/enki/tests/test_plugins (master)$ aptitude show python-sphinx | grep Version
Version: 1.1.3+dfsg-7ubuntu2
```

```
andrei@a-desktop ~/code/enki/tests/test_plugins (master)$ lsb_release -a 
No LSB modules are available.
Distributor ID:	LinuxMint
Description:	Linux Mint 16 Petra
Release:	16
Codename:	petra
```

**Comment by bjones1** [comment]
I didn't realize this would reproduce consistently. Pan, would you try to reproduce using the given versions of CodeChat and Sphinx?

**Comment by panhejia** [comment]
Sorry for the late reply. May I have the version number for sphinx-build? simply run 
```sphinx-build --version```
I believe this is still a sphinx version problem since with the following configuration I still cannot reproduce the error.
```
minttester@vm ~/enki/tests/test_plugins $ aptitude show python-sphinx | grep Version
Version: 1.1.3+dfsg-7ubuntu2
minttester@vm ~/enki/tests/test_plugins $ pip show CodeChat
---
Name: CodeChat
Version: 0.0.18
Location: /usr/local/lib/python2.7/dist-packages
Requires: docutils
minttester@vm ~/enki/tests/test_plugins $ lsb_release -aNo LSB modules are available.
Distributor ID:	LinuxMint
Description:	Linux Mint 16 Petra
Release:	16
Codename:	petra
minttester@vm ~/enki/tests/test_plugins $ sphinx-build --version
Sphinx (sphinx-build) 1.2.3
minttester@vm ~/enki/tests/test_plugins $ 
```
Running run_all.py generates 
```
Ran 275 tests in 105.539s
OK (skipped=66)
```

**Comment by hlamer** [comment]
```
andrei@a-desktop ~/code/enki (locator)$ sphinx-build -v 
Sphinx v1.1.3
```

**Comment by panhejia** [comment]
The following versions have been tested. None can reproduce the above mentioned bug even though segmentation fault or python.exe stops working happened twice.
```
CodeChat sphinx-build systemversion
0.0.17   1.2.3        Win7 64bit
0.0.17   1.1.3        Win7 64bit
0.0.18   1.2.3        Mint 16 32bit
0.0.18   1.1.3        Mint 16 32bit
```

**Comment by hlamer** [comment]
I checked the code and found that the test fails always. It is marked as `expectedFailure`. @bjones1  added `inMainLoop` decorator and the failure became loud.
I've changed the decorators order to suppress the backtrace. So test log is pretty again.

There are 3 tests which are marked as `expectedFailure` and all 3 fail for me. Do we really need this tests?

**Comment by bjones1** [comment]
Ouch. I wish I'd looked at this first :). It did produce a few failures in CodeChat (with Sphinx 1.3) that need to be addressed separately (see https://bitbucket.org/bjones/documentation/issue/30/support-sphinx-13). I added the exception hook to provide a more useful traceback when a test fails. But, looking at it again, I should probably save `sys.exc_info()` then properly re-raise it, rather than just printing a trackback. 

In terms of expected fails, test_sync12 is a reminder to me of missing functionality (sync past images). I've converted that into issue https://github.com/bjones1/enki/issues/42 and will remove it from the code. The two expectedFailures in test_approx_match documents non-intuitive behavior that I hope to fix at some point, but don't have any ideas on how to do that currently. I'd like to leave those where they are as future work.

Unfortunately, I'm in end-of-semester mode until the end of this month, and I won't have much time to work on this. I'll be back at work on this at the beginning of May.
